umap/views.py

```diff
@@ -285,6 +286,36 @@ def get_context_data(self, **kwargs):
 user_dashboard = UserDashboard.as_view()
 
 
+class UserDownload(PaginatorMixin, DetailView, SearchMixin):
+    model = User
+
+    def get_object(self):
+        return self.get_queryset().get(pk=self.request.user.pk)
+
+    def get_maps(self):
+        qs = self.get_search_queryset() or Map.objects.all()
+        qs = qs.filter(owner=self.object).union(qs.filter(editors=self.object))
+        return qs.order_by("-modified_at")
+
+    def render_to_response(self, context, *args, **kwargs):
+        zip_buffer = io.BytesIO()
+        with zipfile.ZipFile(zip_buffer, "a", zipfile.ZIP_DEFLATED, False) as zip_file:
+            for map_ in self.get_maps():
```

Comment:
We might want to put a hard top limit (like 50) to avoid saturating the memory for prolific users but it requires to make it explicit!